### ISSUE TITLE
Docs: serverMainFields should be an array

### ISSUE THREAD

**Issue by sadin999** [remix-run/remix#5980]
### What version of Remix are you using?

1.15.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Minor docs issue - 

Prepare for v2 docs have the following examples: 
```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  publicPath: "/_static/build/",
  serverBuildPath: "server/index.js",
  serverMainFields: "main, module",
  serverModuleFormat: "cjs",
  serverPlatform: "node",
  serverMinify: false,
};
```
`serverMainFields` should be an array - but in the docs, it's a comma separated list. This results in the following when starting the dev server:

```
  errors: [
    {
      id: '',
      pluginName: '',
      text: '"mainFields" must be an array',
```

Setting to array works fine: e.g.
```
serverMainFields: ["main", "module"]
```

Happy to do a docs PR, just wanted to confirm it first.

### Expected Behavior

Dev server should start.

### Actual Behavior

Dev server cannot start.

**Comment by MichaelDeBoey** [comment]
Fixed by #5972

**Comment by SleeplessByte** [comment]
Same issue exists for conditions; docs also seem to not yet have been deployed.
